### ISSUE TITLE
[Bug] LayerExtension has no effect on Tile3DLayer 

### ISSUE THREAD

**Issue by RELNO** [visgl/deck.gl#7952]
### Description

When using LayerExtension with simple frag shader, most layers are impacted, besides Tile3DLayer.
Example below shows `Tile3DLayer`  and `TileLayer`. `LayerExtension` only impacts `TileLayer` (https://codesandbox.io/p/sandbox/small-water-fkrzcm)
**Note: this applies to all Tile3DLayer sources, regardless if its google 3d tiles, archgis, etc.** 

<img width="1218" alt="Screenshot 2023-06-16 at 1 42 04 PM" src="https://github.com/visgl/deck.gl/assets/14296774/504ca0cd-8c88-4ebe-ad06-f6b9d1851427">
<img width="1221" alt="Screenshot 2023-06-16 at 1 42 10 PM" src="https://github.com/visgl/deck.gl/assets/14296774/c2f6de08-db21-4893-9043-4bb51b60bbc2">

Where Extension is:

```
import { LayerExtension } from "@deck.gl/core";

export default class ColorExtension extends LayerExtension {

  getShaders() {
    return {
      inject: {
        "fs:DECKGL_FILTER_COLOR": `
        color = vec4(1,0,0, 1);
            `,
      },
    };
  }
}

```

### Flavors

- [ ] Script tag
- [X] React
- [ ] Python/Jupyter notebook
- [X] MapboxOverlay
- [X] GoogleMapsOverlay
- [ ] CartoLayer
- [X] ArcGIS

### Expected Behavior

LayerExtension frag shader should apply to all layers 

### Steps to Reproduce

add LayerExtension with shader to Tile3DLayer. See here https://codesandbox.io/p/sandbox/small-water-fkrzcm

### Environment

- Framework version: latest, also tested on prev. release.
- Browser: any
- OS: any

**Comment by RELNO** [comment]
@Pessimistress since tile3d is composite of ScenegraphLayer (in this case), I wonder if https://github.com/visgl/deck.gl/pull/5214 might be related.

**Comment by RELNO** [comment]
Further investigation shows that `getShaders` has no impact on extended `Tile3DLayer`:
```
import { Tile3DLayer } from "@deck.gl/geo-layers";

export default class ExtendedTile3DLayer extends Tile3DLayer {
  initializeState(state) {
    super.initializeState();
    console.log("This will be logged");
  }

  getShaders() {
    console.log("This will not be logged")
    return Object.assign({}, super.getShaders, {
      fs: "color = vec4(1.0, 0.0, 0.0, 1.0);",
    });

  getLoadOptions() {
    console.log("This won't as well :( ");
  }
    
  }
}
```

**Comment by Pessimistress** [comment]
@RELNO Regarding your last comment, composite layers never call `getShaders`. The Tile3DLayer renders ScenegraphLayer, which calls its own `getShaders`. Extensions should theoretically work, though, I'll need to investigate.

**Comment by RELNO** [comment]
@Pessimistress thanks, makes total sense. 
In terms of the sublayer (Scenegraph), two things to note: 
- I cloned the tile3d layer source and checked to see if the extension is passed:
```
 {
      key: "_make3DModelLayer",
      value: function _make3DModelLayer(tileHeader) {
        var _tileHeader$content2 = tileHeader.content,
          gltf = _tileHeader$content2.gltf,
          instances = _tileHeader$content2.instances,
          cartographicOrigin = _tileHeader$content2.cartographicOrigin,
          modelMatrix = _tileHeader$content2.modelMatrix;
          // => here we can log the state/props and see that `extensions` are passed 
``` 
- When passing the shader using `LayerExtension` (as mentioned above) to a non-google source (such as archgis) **on the very first load** (i.e, before the texture parser warms up), it seems the shader does get passed down to the layer. This hints that the issue is with the lazy load nature of this data.

**Comment by Pessimistress** [comment]
Fixed in v8.9.20.

**Comment by RELNO** [comment]
works great, thanks @Pessimistress!
